tests/plugins/clipboard_figure.test.ts

```diff
@@ -128,6 +128,22 @@ describe.each(["chart", "image"])("Clipboard for %s figures", (type: string) =>
     expect(model.getters.getFigures(sheetId)).toHaveLength(0);
   });
 
+  test("Figure is copied to edge of the sheet", () => {
+    model.dispatch("UPDATE_FIGURE", {
+      sheetId,
+      id: figureId,
+      height: 256,
+      width: 257,
+    });
+    model.dispatch("SELECT_FIGURE", { id: figureId });
+    copy(model);
+    paste(model, "Z100");
+    const copiedFigure = model.getters.getFigure(sheetId, getCopiedFigureId())!;
+    const { height, width } = model.getters.getMainViewportRect();
+    expect(copiedFigure.x).toBe(width - copiedFigure.width - 96);
+    expect(copiedFigure.y).toBe(height - copiedFigure.height - 28);
```

Comment:
Adapt the test to reflect your new computations - no viewport size + no  96/28 constants